internal/kubernetes/operator/deployment/deployment_test.go

```diff
@@ -38,6 +38,11 @@ type MockAtlasOperatorClusterStore struct {
 	projectToClusterToSchedule            map[string]map[string]*mongodbatlas.CloudProviderSnapshotBackupPolicy
 	projectToServerlessClusters           map[string]map[string]*mongodbatlas.Cluster
 	projectIDToServerlessPrivateEndpoints map[string]map[string][]mongodbatlas.ServerlessPrivateEndpointConnection
+	projectIDToGlobalCluster              map[string]map[string]*mongodbatlas.GlobalCluster
+}
+
```

Comment:
[q] Can you update the e2e test as well?

Reply:
Some errors with project deletion. 	Error: DELETE https://cloud-dev.mongodb.com/api/public/v1.0/groups/1234: 500 (request "UNEXPECTED_ERROR") Unexpected error.
[2023/01/09 10:12:12.802]     projects_test.go:121: got="Error: DELETE https://cloud-dev.mongodb.com/api/public/v1.0/groups/1234: 500 (request \"UNEXPECTED_ERROR\") Unexpected error.\n"

Reply:
No, the project deletion error are not related to the library upgrade but to some endpoint in cloud-dev that are not working as expected. You should still be able to update the e2e test [kubernetes_config_generate_test.go](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/kubernetes_config_generate_test.go)

Reply:
Global cluster is a feature for advanced and legacy clusters. And as I see, we do not write e2e tests for each feature.

Reply:
at this line [kubernetes_config_generate_test](https://github.com/mongodb/mongodb-atlas-cli/blob/d72fcee764cdd38f3831ae10b93d13ffb9d04157/test/e2e/atlas/kubernetes_config_generate_test.go#L70), we are setting up a cluster for the e2e test so you should be able to test that the `kub generate command` has created the correct yaml for the cluster right?

Reply:
Yeah. But now our tests compare only the names of the resources. I agree that we should add e2e for each feature, but let's do it in another PR. Do you mind?

Reply:
SGTM. Could you create a ticket if you are not planning to address this as part of [CLOUDP-152961](https://jira.mongodb.org/browse/CLOUDP-152961)? Thanks!

Reply:
Done. [CLOUDP-153348](https://jira.mongodb.org/browse/CLOUDP-153348)